packages/java-edition/src/binder/index.ts

```diff
@@ -226,41 +274,7 @@ export function dissectUri(uri: string, ctx: UriBinderContext) {
 		if (pack !== 'data' && pack !== 'assets') {
 			continue
 		}
-		const candidateResources: [Resource, string][] = []
-		if (rest.length === 1) {
-			const resources = Resources.get('')
-			for (const res of resources ?? []) {
-				if (res.pack !== pack) {
-					continue
-				}
-				let identifier = rest[0]
-				if (!identifier.endsWith(res.ext)) {
-					continue
-				}
-				identifier = identifier.slice(0, -res.ext.length)
-				if (res.identifier && identifier !== res.identifier) {
-					continue
-				}
-				candidateResources.push([res, identifier])
-			}
-		}
-		for (let i = 1; i < rest.length; i += 1) {
-			const resources = Resources.get(rest.slice(0, i).join('/'))
-			for (const res of resources ?? []) {
-				if (res.pack !== pack) {
-					continue
-				}
-				let identifier = rest.slice(i).join('/')
-				if (!identifier.endsWith(res.ext)) {
-					continue
-				}
-				identifier = identifier.slice(0, -res.ext.length)
-				if (res.identifier && identifier !== res.identifier) {
-					continue
-				}
-				candidateResources.push([res, identifier])
-			}
-		}
+		const candidateResources = getCandidateResourcesForRel(rel)
```

Comment:
The logic between `const parts = res.split('/')` and here is also duplicated in `getCandidateResourcesForRel`

Reply:
Hmm I see. Though it already returns a list of two things: `function getCandidateResourcesForRel(rel: string): [res: Resource, identifier: string][]`, so maybe it wouldn't be that bad to have it also return the namespace in that tuple (or change to an object)?